st4/__init__.py

```diff
@@ -66,6 +72,34 @@ def m_metals_status(self, params: Any) -> None:
         else:
             session.erase_window_status_async(key)
 
+    def m_metals_publishDecorations(self, decorationsParams: Any) -> None:
+        if not isinstance(decorationsParams, dict):
+            return
+
+        session = self.weaksession()
+        if not session:
+            return
+
+        uri = decorationsParams.get('uri')
+        if not uri:
+            return
+
+        session_buffer = session.get_session_buffer_for_uri_async(uri)
+        if not session_buffer:
+            print("no session builder")
+            return
+
+        for sv in session_buffer.session_views:
+            if(sv.view.file_name() == uri_to_filename(uri)):
```

Comment:
This should always be true :) we don't need this `if` check.

Reply:
there is no guarantee that the URI in `metals/publishDecorations` is about the current view file.

Reply:
We already found the relevant buffer with find_session_buffer_for_uri_async. All attached views are guaranteed to have the correct file name. If that’s not the case then that’s a bug in LSP (but I’m pretty sure this has been working correctly for ages now).